chrome-launcher/chrome-launcher.ts

```diff
@@ -88,18 +96,24 @@ export class Launcher {
   }
 
   private get flags() {
-    const flags = DEFAULT_FLAGS.concat([
-      `--remote-debugging-port=${this.port}`,
+    // Event if a user ignores default flags these flags are required to do a
+    // proper lifecycle cleanup as well as to drive chrome.
+    const requiredFlags = [
       // Place Chrome profile in a custom location we'll rm -rf later
-      `--user-data-dir=${this.userDataDir}`
-    ]);
+      `--remote-debugging-port=${this.port}`, `--user-data-dir=${this.userDataDir}`
+    ];
+
+    if (this.opts.noDefaultFlags === true) {
+      return requiredFlags.concat([this.startingUrl]);
+    }
+
+    const flags = DEFAULT_FLAGS.concat(requiredFlags);
```

Comment:
order wise you're putting the required ones last, which i wouldn't expect. 

for consistency lets always do `.push()` and ignore `concat()`.